query-planner/src/builder.rs

```diff
@@ -777,3 +765,19 @@ fn flat_wrap(kind: NodeCollectionKind, mut nodes: Vec<PlanNode>) -> PlanNode {
         }
     }
 }
+
+fn maybe_auto_fragmentization<'a, 'q: 'a>(
+    context: &'q QueryPlanningContext<'q>,
+    selection_set: SelectionSetRef<'q>,
+) -> (String, SelectionSetRef<'a>) {
```

Comment:
Nit: just for consistency, can we do the printing/minification of both frags and selection set in the same place? So we'd either return `(String, String)` or (not sure if syntax is right, but...) `([FragmentDefinitionRef<'a>], SelectionSetRef<'a>)`